unyt/mpl_interface.py

```diff
@@ -47,24 +47,31 @@ def axisinfo(unit, axis):
             """
             if isinstance(unit, tuple):
                 unit = unit[0]
-            unit_obj = Unit(unit)
+            unit_obj = unit if isinstance(unit, Unit) else Unit(unit)
+            name = getattr(axis, "_unytarrayname", "")
             if unit_obj.is_dimensionless:
-                label = ""
+                label = name
             else:
+                name += " "
                 unit_str = unit_obj.latex_representation()
                 if unyt_arrayConverter._labelstyle == "[]":
-                    label = "$\\left[" + unit_str + "\\right]$"
+                    label = name + "$\\left[" + unit_str + "\\right]$"
```

Comment:
Probably want a space between `name` and unit here.

Reply:
there is, the space is added prior to this line